endpoint/domain_filter.go

```diff
@@ -78,9 +95,34 @@ func matchFilter(filters []string, domain string, emptyval bool) bool {
 	return false
 }
 
+// matchRegex determines if a domain matches the configured regular expressions in the DomainFilter.
+// The negativeRegex, if set, takes precedence over regex. Therefore,
+// matchRegex returns true when only regex regular expression matches the domain.
+// Otherwise, if either negativeRegex matches or regex does not match the domain, it will return false.
+func matchRegex(regex string, negativeRegex string, domain string) bool {
+	strippedDomain := strings.ToLower(strings.TrimSuffix(domain, "."))
+
+	if negativeRegex != "" {
+		match, err := regexp.MatchString(negativeRegex, strippedDomain)
+		if err != nil {
+			log.Errorf("Failed to filter domain %s with the regex-exclusion filter: %v", domain, err)
+		}
+		if match {
+			return false
+		}
+	}
+	match, err := regexp.MatchString(regex, strippedDomain)
+	if err != nil {
+		log.Errorf("Failed to filter domain %s with the regex filter: %v", domain, err)
```

Comment:
Same as above, we need to handle the error.

Reply:
Same as above